### ISSUE TITLE
Default value incompatible with depends

### ISSUE THREAD

**Issue by sabberworm** [c4spar/deno-cliffy#3]
Given an option with both `depends` and `default` value like so:

```typescript
await new Command()
    .option( '--flag1', 'flag 1' )
    .option( '--flag2 <val:string>', 'flag 2', {depends: ['flag1'], default: 'example'} )
    .parse( Deno.args );
```

Cliffy thinks `flag2` is always set and complains:
```
Option --flag2 depends on option: --flag1
```

whenever `flag1` is missing even though `flag2` only has its default value set.

**Comment by c4spar** [comment]
Thx for the issue. I will take a look on it.

**Comment by c4spar** [comment]
Hi @sabberworm, the issue should be fixed in v0.5.1.
Let me know if there is still an issue.
